diff --git a/src/core/backend_mgr.cc b/src/core/backend_mgr.cc
index da6aa9aa..764482ae 100644
--- a/src/core/backend_mgr.cc
+++ b/src/core/backend_mgr.cc
@@ -116,9 +116,20 @@ PythonEnvPath GetPythonEnvPaths()
     static const std::string pythonLibs = (pythonModulesBaseDir / "python311.zip").generic_string();
     static const std::string pythonUserLibs = (pythonModulesBaseDir / "Lib" / "site-packages").generic_string();
 #elif __linux__
-    static const std::string pythonPath = GetEnv("LIBPYTHON_BUILTIN_MODULES_PATH");
-    static const std::string pythonLibs = GetEnv("LIBPYTHON_BUILTIN_MODULES_DLL_PATH");
-    static const std::string pythonUserLibs = (pythonModulesBaseDir / "lib" / "python3.11" / "site-packages").generic_string();
+    static const std::string pythonPath =
+        !GetEnv("MILLENNIUM_PYTHON_HOME").empty()
+            ? GetEnv("MILLENNIUM_PYTHON_HOME")
+            : GetEnv("LIBPYTHON_BUILTIN_MODULES_PATH");
+
+    static const std::string pythonLibs =
+        !GetEnv("MILLENNIUM_PYTHON_STDLIB").empty()
+            ? GetEnv("MILLENNIUM_PYTHON_STDLIB")
+            : GetEnv("LIBPYTHON_BUILTIN_MODULES_DLL_PATH");
+
+    static const std::string pythonUserLibs =
+        !GetEnv("MILLENNIUM_PYTHON_SITEPACKAGES").empty()
+            ? GetEnv("MILLENNIUM_PYTHON_SITEPACKAGES")
+            : (pythonModulesBaseDir / "lib" / "python3.11" / "site-packages").generic_string();
 #elif __APPLE__
     static const std::string pythonPath = GetEnv("LIBPYTHON_BUILTIN_MODULES_PATH");
     static const std::string pythonLibs = GetEnv("LIBPYTHON_BUILTIN_MODULES_DLL_PATH");
